src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -81,6 +81,18 @@ module Make (Inputs : Inputs_intf.S) = struct
           *)
     }
 
+  (** Type for an account that wasn't yet hashed.
```

Comment:
General question regarding your comment from the PR description:

>When an account is touched in a few transactions of the same block, it's gets hashed a few times, whereas in fact only the final hash is truly needed

I'm wondering if this does not change the original application logic. Assume tx1 changes A and B, and tx2 changes C and A. So A1 becomes A2 and then A3, B1 becomes B2, C1 becomes C2. Surely, after tx1 and tx2 the final account state for A is A3, but don't we /at all/ care about the intermediate Merkle tree? Can we just rehash the MT based on A3 B2 C2? 

(I suspect yes, just double checking)